packages/mephisto-task/src/live.js

```diff
@@ -24,27 +24,67 @@ const AGENT_STATUS = {
   MEPHISTO_DISCONNECT: "mephisto disconnect",
 };
 
+const STATUS_TO_TEXT_MAP = {
+  [AGENT_STATUS.EXPIRED]:
+    "This task is no longer available to be completed. " +
+    " Please return it and try a different task",
+  [AGENT_STATUS.TIMEOUT]:
+    "You took to long to respond to this task, and have timed out. " +
+    "The task is no longer available, please return it.",
+  [AGENT_STATUS.DISCONNECT]:
+    "You have disconnected from our server during the duration of the task. " +
+    "If you have done substantial work, please reach out to see if we can recover it. ",
+  [AGENT_STATUS.RETURNED]:
+    "You have disconnected from our server during the duration of the task. " +
+    "If you have done substantial work, please reach out to see if we can recover it. ",
+  [AGENT_STATUS.PARTNER_DISCONNECT]:
+    "One of your partners has disconnected while working on this task. We won't penalize " +
+    "you for them leaving, so please submit this task as is.",
+  [AGENT_STATUS.MEPHISTO_DISCONNECT]:
+    "Our server appears to have gone down during the " +
+    "duration of this Task. Please send us a message if you've done " +
+    "substantial work and we can find out if the task is complete enough to " +
+    "compensate.",
+};
+
 const useMephistoLiveTask = function (props) {
   const [connectionStatus, setConnectionStatus] = React.useState("");
-  const [agentState, setAgentState] = React.useState(null);
   const [agentStatus, setAgentStatus] = React.useState(null);
 
   const defaultConnectionHook = useMephistoSocket;
   const useConnectionHook = props.customConnectionHook || defaultConnectionHook;
 
   const taskProps = useMephistoTask();
+
+  const { initialTaskData } = taskProps;
+
+  function onMessageReceived(message) {
+    const { state, message_id, timestamp, ...remaining_message } = message;
+    if (state !== undefined) {
+      props.onStateUpdate && props.onStateUpdate({ state });
+    }
+    if (Object.keys(remaining_message).length != 0) {
+      props.onMessageReceived && props.onMessageReceived(remaining_message);
+    }
+  }
+
+  React.useEffect(() => {
+    if (initialTaskData?.raw_messages) {
+      for (const raw_message of initialTaskData.raw_messages) {
+        onMessageReceived(raw_message);
+      }
+    }
+  }, [initialTaskData]);
+
   const socketProps = useConnectionHook({
     onConnectionStatusChange: (connectionStatus) => {
       setConnectionStatus(connectionStatus);
     },
-    onStateUpdate: ({ state, status }) => {
-      setAgentState(state);
+    onStatusUpdate: ({ status }) => {
       setAgentStatus(status);
```

Comment:
Really happy about this simplification and consolidation. States and status always felt a little confusing to me.